src/snowflake/snowpark/_internal/data_source/drivers/base_driver.py

```diff
@@ -58,3 +102,13 @@ def validate_numeric_precision_scale(
         elif scale is not None:
             return False
         return True
+
+
+def get_dbms_packages(dbapi2_conn) -> Union[list, None]:
+    from snowflake.snowpark._internal.data_source.utils import (
+        detect_dbms,
+        UDTF_PACKAGE_MAP,
+    )
+
+    dbms_type, _ = detect_dbms(dbapi2_conn)
+    return UDTF_PACKAGE_MAP.get(dbms_type)
```

Comment:
I think this function can be moved to utils modules?

Reply:
there will be circular import if I move it to utils